pkg/task/task.go

```diff
@@ -147,3 +222,41 @@ type CheckpointInfo struct {
 type CompletedTask struct {
 	Checkpoint TaskCheckpoint `json:"checkpoint"`
 }
+
+func newCheckpointInfo() CheckpointInfo {
+	return CheckpointInfo{
+		CompletedTasks: make(map[string]*CompletedTask),
+	}
+}
+
+func (c CheckpointInfo) taskCompleted(name string, completedTask *CompletedTask) {
+	c.CompletedTasks[name] = completedTask
+}
+
+func getCheckpointFile(file string) (*CheckpointInfo, error) {
+	logger.V(4).Info("Reading checkpoint", "file", file)
+	content, err := os.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed reading checkpoint file: %v\n", err)
+	}
+	checkpointInfo := &CheckpointInfo{}
+	err = yaml.Unmarshal(content, checkpointInfo)
+	if err != nil {
+		return nil, fmt.Errorf("failed unmarshalling checkpoint: %v\n", err)
+	}
+
+	return checkpointInfo, nil
+}
+
+/* When reading from a yaml file, there isn't a direct way in Go to do a type conversion from interface{} to the desired type.
+We use interface{} because the TaskCheckpoint type will vary depending on what's needed for a specific task. The known workaround
+for this is to marshal & unmarshal it into the checkpoint type. This UnmarshalTaskCheckpoint method marshals the received task checkpoint (type interface{})
+then unmarshalls it into the desired type.
+*/
+func UnmarshalTaskCheckpoint(taskCheckpoint TaskCheckpoint, config TaskCheckpoint) error {
```

Comment:
I was referring to this:
```suggestion
/* UnmarshalTaskCheckpoint unmarshals bla bla bla bla
When reading from a yaml file, there isn't a direct way in Go to do a type conversion from interface{} to the desired type.
We use interface{} because the TaskCheckpoint type will vary depending on what's needed for a specific task. The known workaround
for this is to marshal & unmarshal it into the checkpoint type. This UnmarshalTaskCheckpoint method marshals the received task checkpoint (type interface{})
then unmarshalls it into the desired type.
*/
func UnmarshalTaskCheckpoint(taskCheckpoint TaskCheckpoint, config TaskCheckpoint) error {
```